### DISCUSSION TITLE
1.52

### DISCUSSION THREAD

**Discussion by alex19EP** [espeak-ng/espeak-ng#2060] [Announcements]
## changes

**attention!** this is the last release witch support the autoconf build system. use cmake to compile espeak-ng.

*  added cmake build system  -- djphoenix
*  Added stress marks to phoneme events -- Rotem Dan

### bug fixes

*  a meny fixes to core C code -- djphoenix

### android

*  Added directBoot support -- beqabeqa473
*  Integrated ndk-build step into main gradle pipeline

updated languages:
*  ba (Bashkir) -- Andiv06
*  be (Belarusian) -- Andiv06, bespsm, Alaksiej Stankievič
*  cmn (Mandarin) -- Cameron Wong
*  en (English) -- Bill Dengler
*  es (Spanish) -- Sukil Etxenike, Rmcpantoja, gregodejesus
*  fa (Persian) -- MH
*  gn (Guarani) -- Andiv06
*  he (Hebrew) -- Omer I.S, Andiv06
*  it (Italian) -- Christian Leo Mameli
*  ja (Japanese) -- Karl Eick
*  kl (Greenlandic) -- Andiv06
*  ky (Kyrgyz) -- Andiv06, Valdis Vitolins
*  lv (Latvian) -- Valdis Vitolins
*  nog (Nogai) -- Andiv06, boracasli98
*  qdb (Lang Belta) -- Andiv06
*  qya (Quenya) -- Andiv06
*  quc (K'iche') -- Andiv06
*  ro (Romanian) -- Andiv06
*  ru (Russian) -- Kirill-jjj, a11cf0
*  shn (Shan Tay Yai) -- ronaldaug
*  tk (Turkmen) -- Andiv06
*  tr (Turkish) -- Gurbuzguven, Qchristensen
*  uk (Ukrainian) -- Andrij Mizyk
*  uz (Uzbek) -- Andiv06

new languages:
*  en-Shaw (English, Shavian script) -- Luna Rose
*  ti (Tigrinya) -- Biniam Gebremichael
*  mto (Totontepec Mixe) -- Bill Dengler, Elizabeth Resendiz
*  fo (Faroese) -- Andras Eliassen, iSolveIT ApS and Setur.fo/Ravnur
*  kaa (Karakalpak) -- atabekm
*  xex (Xextan) -- Filomena Rocca

## New Contributors

* @sukiletxe made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1167
* @seanbudd made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1211
* @ryandesign made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1003
* @Dekedro made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1175
* @kmamadoudram made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1178
* @MH140000 made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1235
* @TheQuinbox made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1254
* @Gurbuzguven made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1260
* @FelipeZanabria made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1292
* @kirill-jjj made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1350
* @djphoenix made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1262
* @bkmgit made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1376
* @hgneng made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1404
* @fireneat made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1413
* @beqabeqa473 made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1423
* @bespsm made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1417
* @PatrykMis made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1449
* @AbdulSamadMalik made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1442
* @andmizyk made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1480
* @rmcpantoja made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1517
* @Idorobots made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1696
* @mueller-ma made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1720
* @DimitryAndric made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1722
* @Benjamin-Loison made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1783
* @tuxpoldo made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1811
* @lukaszgo1 made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1813
* @heitbaum made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1858
* @kenta2 made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1848
* @biniamgebremichael made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1845
* @hammera made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1882
* @SimpleMango made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1670
* @cimm made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1682
* @CWBudde made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1757
* @BobSquarePants made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1922
* @eltociear made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1910
* @raphaelmerx made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1879
* @lpmi-13 made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1906
* @danemadsen made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1878
* @atabekm made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1894
* @trenslow made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1929
* @isolveit-aps made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1914
* @WallFlowerWeed made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1881
* @danielwinkler made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1901
* @omeritzics made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1919
* @jtojnar made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1948
* @c-armentano made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1681
* @WilliamTambellini made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1982
* @alanc made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/2008
* @fabiolimace made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/2025
* @cclauss made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1969
* @aherrou made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/2044
* @ehsan-e made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/2020
* @PaulBoersma made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1973
* @thewh1teagle made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1983
* @Danstiv made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1965
* @wilt00 made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1946
* @michal-bida made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1931
* @lunarna-gh made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1924
* @Sur3 made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1916
* @rotemdan made their first contribution in https://github.com/espeak-ng/espeak-ng/pull/1700

**Full Changelog**: https://github.com/espeak-ng/espeak-ng/compare/1.51...1.52

<hr /><em>This discussion was created from the release <a href='https://github.com/espeak-ng/espeak-ng/releases/tag/1.52.0'>1.52</a>.</em>

**Comment by hammera** [comment]
@alex19EP , congratulate the new release.
This day afternoon I wanted opening a new PR with hu_list and hu_rules file related before go to holiday, this is too late, release comed out I think since 1 hour ago.
I few minute previous seed only the e-mail notification. :-(:-(
PR number is #2061.
Fortunatelly hu_list file contains only one change with missed the new stable release, hu_rules file contains only 56 new added exceptions.
Have possibility future put this changes an 1.52 maintenance release, or have possibility to merge #2061 related changes only with master branch?
If only the master branch merge are acceptable, I agree this.
I looked at the [espeak-ng milestone list](https://github.com/espeak-ng/espeak-ng/milestones) last night, but I didn't see a specific target date for the 1.52 release at estimated 22:00 PM.

Attila

**Reply by crazy-peiqi** [reply]
语音合成不能合成中文，合成中文就出现pinyin合成

**Comment by mh292929** [comment]
Hi. Thanks but why fa_extra doesn't compile in new version?

**Reply by alex19EP** [reply]
hmm. can't reproduce. please provide more info.

**Comment by mh292929** [comment]
I added sum words in fa_extra dictionary but now isn't in this build.

**Comment by mh292929** [comment]
hello I removed all the strings from fa_extra and moved them to fa_list because they were not included in the last version.

**Comment by milekpl** [comment]
Why is there no Windows build?

**Reply by wilt00** [reply]
Nightly Windows builds can be downloaded from the CI job here: https://github.com/espeak-ng/espeak-ng/actions/workflows/windows-msbuild.yml?query=event%3Apush+is%3Asuccess+branch%3Amaster - specifically, the build job for the stable 1.52.0 release is here: https://github.com/espeak-ng/espeak-ng/actions/runs/12297400030 , where you can download the 'windows-Release-msi' installer. I agree that it would be nice to have that build included in the release here, although just getting the release train rolling again after three years of accumulated changes is accomplishment enough!

**Reply by alex19EP** [reply]
hello. I added this artifact to release. I'll try to automate the addition of the installer for Windows before the next release.

**Comment by seanbudd** [comment]
It's great to see another stable release! Is this going to be a regular thing, and do you have any idea on the frequency?
Trying to figure if we should switch to using stable releases rather than master commits.

**Comment by WilliamTambellini** [comment]
tks @alex19EP

**Comment by AcTePuKc** [comment]
Hey everyone,  

Could you update the Windows installer to **automatically set** the necessary environment variables for Phonemizer integration?  

Currently, users have to manually configure the following for Phonemizer to detect eSpeak NG properly:  

```
[System.Environment]::SetEnvironmentVariable("PHONEMIZER_ESPEAK_PATH", "$INSTALL_DIR", [System.EnvironmentVariableTarget]::Machine)  
[System.Environment]::SetEnvironmentVariable("PHONEMIZER_ESPEAK_LIBRARY", "$INSTALL_DIR\libespeak-ng.dll", [System.EnvironmentVariableTarget]::Machine)  
```  

where `$INSTALL_DIR` is the eSpeak NG installation directory.  

The official documentation suggests adding eSpeak NG to `PATH`, but this alone isn't sufficient for Phonemizer. Automating these environment variables during installation would greatly simplify setup for Windows users.  

Would it be possible to add this step to the **Windows installer** so users don’t have to set it manually or search for workarounds? It would make the installation process much smoother.  

Thanks a lot!

**Reply by alex19EP** [reply]
hello. pr to do that is welcome

**Comment by crazy-peiqi** [comment]
不能语音合成中午，不知道是不是mis的安装包有点问题

**Reply by wudidiandian** [reply]
win11上我都不能安装成功espeak-ng.msi包，双击没有任何反应，好奇怪

**Comment by six168** [comment]
Why is there no linux build?

**Comment by uyt3ar** [comment]
Hello
@alex19EP Isn't it time to release a new version now? This version has been around for about two years now.

**Comment by hammera** [comment]
@alex19EP, before 1.52 release, don't forgot to merge the hungarian #2509 PR to the main branch.

Attila
